internal/encryption/keyprovider.go

```diff
@@ -73,15 +144,12 @@ func (e *targetBuilder) setupKeyProvider(cfg config.KeyProviderConfig, stack []c
 	for _, s := range stack {
 		if s == cfg {
 			addr, diags := keyprovider.NewAddr(cfg.Type, cfg.Name)
-			diags = diags.Append(
-				&hcl.Diagnostic{
-					Severity: hcl.DiagError,
-					Summary:  "Circular reference detected",
-					// TODO add the stack trace to the detail message
-					Detail: fmt.Sprintf("Can not load %q due to circular reference", addr),
-				},
-			)
-			return diags
+			return diags.Append(&hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "Circular reference detected",
+				// TODO add the stack trace to the detail message
+				Detail: fmt.Sprintf("Can not load %q due to circular reference", addr),
```

Comment:
```suggestion
				Detail: fmt.Sprintf("Cannot load %s due to circular reference between key providers.", addr),
```

Along with being a little more specific about what is relevant to this circular reference and adding the period to the end to make it a full sentence, I also changed `%q` to `%s` because it seems like `keyprovider.Addr` represents a reference address like `key_provider.foo.bar` and we don't typically quote those in error messages once we've confirmed that they refer to something valid.

(There are some other "can not" vs. "cannot" typos elsewhere in this function too, FWIW.)